.github/workflows/actions_build.yml

```diff
@@ -0,0 +1,145 @@
+
+    - if: ${{ matrix.coverage }}
+      uses: codecov/codecov-action@v1
+
+    - name: Collecting the test reports ..
+      if: failure()
+      run: find . '(' -name 'hs_err_*.log' -or -path '*/build/reports/tests' ')' -exec tar rf "reports-JVM-${{ matrix.java }}.tar" {} ';'
+
+    - name: 'Upload Artifact'
+      if: failure()
+      uses: actions/upload-artifact@v2
+      with:
+        name: reports-JVM-${{ matrix.java }}
+        path: reports-JVM-${{ matrix.java }}.tar
+        retention-days: 3
+
+  build_on_other:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ macos-latest, windows-latest]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up JDK 15
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '15'
+
+      - uses: burrunan/gradle-cache-action@v1.10
+        with:
+          remote-build-cache-proxy-enabled: false
+          arguments: --no-daemon --stacktrace -PnoLint -PnoWeb :core:test :grpc:test :thrift0.13:test :it:server:test
+
+      - name: Collecting the test reports ..
+        if: failure()
+        run: find . '(' -name 'hs_err_*.log' -or -path '*/build/reports/tests' ')' -exec tar rf "reports-${{ matrix.os }}.tar" {} ';'
+
+      - name: 'Upload Artifact'
+        if: failure()
+        uses: actions/upload-artifact@v2
+        with:
+          name: reports-${{ matrix.os }}
+          path: reports-${{ matrix.os }}.tar
+          retention-days: 3
+
+  build_site:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up JDK 15
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '15'
+
+      - name: Install svgbob_cli
+        run: sudo apt install cargo && cargo install svgbob_cli
+
+      - uses: burrunan/gradle-cache-action@v1.10
+        with:
+          remote-build-cache-proxy-enabled: false
+          arguments: --no-daemon --stacktrace --parallel --max-workers=4 :site:siteLint :site:site
+
+      - name: Collecting the test reports ..
+        if: failure()
+        run: find . '(' -name 'hs_err_*.log' -or -path '*/build/reports/tests' ')' -exec tar rf "reports-site.tar" {} ';'
+
+      - name: 'Upload Artifact'
+        if: failure()
+        uses: actions/upload-artifact@v2
+        with:
+          name: reports-site
+          path: reports-site.tar
+          retention-days: 3
+
+  build_leak:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up JDK 15
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '15'
+
+      - uses: burrunan/gradle-cache-action@v1.10
+        with:
+          remote-build-cache-proxy-enabled: false
+          arguments: --no-daemon --stacktrace --parallel --max-workers=4 -Pleak -PnoLint test
```

Comment:
Why separate job for leak test?

Reply:
We have a different job for leak test https://github.com/line/armeria/blob/master/.appveyor.yml#L17
so that we can reduce the amount of time for checking the leak.
Do you think we need to run the leak build for all JDK versions and Operating Systems?

Reply:
Don't think so but you can use a conditional to enable it only on one JDK / OS in the normal build. It seems that this will mostly be redundant with the normal build

Reply:
Yeah, let me include the leak build to the JDK 15 build.
It probably does not take too long. 😄